src/mpid/ch4/src/ch4r_callbacks.c

```diff
@@ -553,8 +553,12 @@ int MPIDIG_send_long_req_target_msg_cb(int handler_id, void *am_hdr, void **data
         }
         MPID_THREAD_CS_EXIT(VCI, MPIDIU_THREAD_MPIDIG_GLOBAL_MUTEX);
     } else {
-        /* Matching receive was posted, tell the netmod */
-        MPIR_Comm_release(root_comm);   /* -1 for posted_list */
```

Comment:
You removed the `MPIR_Comm_release` here, but where is the release actually going to happen now? Does that happen automatically when the request is freed?

Reply:
It is in the comments I added below. 

What actually happens is: we do `MPIR_Comm_release` for removing from `posted_list`, then we need do `MPIR_Comm_add_ref`  for attaching the comm to `rreq`. The latter is necessary to fix a bug when user free the comm before we do `ack` later. 

Now release the reference then immediately add the reference back cancels, so the change appears to be just removing `MPIR_Comm_release`.